src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -270,24 +270,73 @@ class DataCube {
     extendFinalDimensionWithData(axis, dataToAdd) {
         const finalShape = this.getDataShape()
         finalShape[axis]++;
-        
-        const finalData = this.insertIntoFinalDimension(dataToAdd, finalShape[axis], finalShape[axis] - 1)     
+
+        const finalData = this.insertIntoFinalDimension(dataToAdd, finalShape[axis], finalShape[axis] - 1)
         this.data = ndarray(finalData, finalShape)
     }
 
     insertIntoFinalDimension(dataToAdd, dimensionSize, locationInDimension) {
         const dataArr = this.data.data;
         for (let i = 0; i < dataToAdd.length; i++) {
             dataArr.splice(
-                i * dimensionSize + locationInDimension, 
-                0, 
+                i * dimensionSize + locationInDimension,
+                0,
                 dataToAdd[i]
             );
         }
 
         return dataArr;
     }
 
+    // axis: integer, dimension axis
+    // dataToAdd: array, values to insert in dimension.
+    // locationInDimension: integer, offset from start of dimension, max length of dimension. Default 0.
+    insertIntoDimension(axis, dataToAdd, locationInDimension = 0) {
```

Comment:
it would probably make sense to add tests for newly added functions.fftr

Reply:
Added tests for `insertIntoDimension` and `setInDimension`.